### PR TITLE
fix in-cluster operator fetch remote chart issue.

### LINKED ISSUE #27719: installPackagePath setting to URL is not working
**Bug description**
When install istio with setting installPackagePath to remote URL, istio operator return "no such file or directory" error.

Example error message:
error   failed to generate IstioOperator spec, failed to read profiles: failed to read profiles: open https:/github.com/istio/istio/releases/download/1.7.0/istio-1.7.0-linux.tar.gz/profiles: no such file or directory

[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Istio operator should download remote tar then unzip and find the charts for installation.

**Steps to reproduce the bug**
[Install istio 1.7.0 as example]
```
curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.7.0 sh -
cd istio-1.7.0
helm template manifests/charts/istio-operator \
  --set hub=docker.io/istio \
  --set tag=1.7.0 \
  --set operatorNamespace=istio-operator \
  --set istioNamespace=istio-system | kubectl apply -f -

kubectl apply -f - <<EOF
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: example-istiocontrolplane
spec:
  profile: default
  **installPackagePath**: https://github.com/istio/istio/releases/download/1.7.0/istio-1.7.0-linux-amd64.tar.gz
EOF

[logs from istio operator pod]
2020-10-03T04:49:27.994773Z     info    leader election cm: istio-operator-lock
2020-10-03T04:49:27.995374Z     info    ControlZ available at 127.0.0.1:9876
2020-10-03T04:49:28.750429Z     info    Registering Components.
2020-10-03T04:49:28.750515Z     info    Adding controller for IstioOperator
2020-10-03T04:49:28.750644Z     info    Controller added
2020-10-03T04:49:28.750656Z     info    Starting the Cmd.
2020-10-03T04:49:28.750973Z     info    attempting to acquire leader lease  istio-operator/istio-operator-lock...
2020-10-03T04:49:28.773450Z     info    successfully acquired lease istio-operator/istio-operator-lock
2020-10-03T04:49:39.456771Z     info    Reconciling IstioOperator
2020-10-03T04:49:39.457118Z     info    Adding finalizer istio-finalizer.install.istio.io to istio-system/example-istiocontrolplane
2020-10-03T04:49:39.464219Z     info    Updating IstioOperator
2020-10-03T04:49:39.464296Z     error   failed to generate IstioOperator spec, failed to read profiles: failed to read profiles: open https:/github.com/istio/istio/releases/download/1.7.0/istio-1.7.0-linux.tar.gz/profiles: no such file or directory
```

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
kubectl version
Client Version: v1.18.8
Server Version: v1.18.6

helm version
version.BuildInfo{Version:"v3.3.0", GitCommit:"8a4aeec08d67a7b84472007529e8097ec3742105", GitTreeState:"dirty", GoVersion:"go1.14.7"}

**How was Istio installed?**
Using istio operator

**Environment where bug was observed (cloud vendor, OS, etc)**
Ubuntu

### PR DESCRIPTION
**Pull request by morvencao**
Please provide a description for what this PR is for.
resolve: https://github.com/istio/istio/issues/27719

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/istio-operator/files/gen-operator.yaml [hunk 1]
```diff
@@ -211,3 +211,9 @@ spec:
               value: "300s"
             - name: REVISION
               value: ""
+          volumeMounts:
+          - mountPath: /tmp
```
Comment (sdake) [inline manifests/charts/istio-operator/files/gen-operator.yaml:215]: please use a different mount path then /tmp.

File: manifests/charts/istio-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -48,4 +48,10 @@ spec:
               value: {{.Values.waitForResourcesTimeout | quote}}
             - name: REVISION
               value: {{.Values.revision | quote}}
+          volumeMounts:
+          - mountPath: /tmp
```
Comment (sdake) [inline manifests/charts/istio-operator/templates/deployment.yaml:52]: same comment about /tmp.

**Review by sdake — CHANGES_REQUESTED** [review]
this PR looks good, although I am concerned it could be used to extract secrets or other critical information from the host filesystem.

Could you make this remote only?

File: manifests/charts/istio-operator/files/gen-operator.yaml (hunk 1 above)
Comment (morvencao) [reply manifests/charts/istio-operator/files/gen-operator.yaml:215]: @sdake The url fetcher by default extrats the chart tgz to `/tmp` dir
https://github.com/istio/istio/blob/83cfd682cdfd9a536a03531aab26174745087087/operator/pkg/helm/urlfetcher.go#L59-L61

**Pull request closed without merging**